systemtests/test_env.py

```diff
@@ -40,75 +89,51 @@ def test_env(ctx, env):
+    * GCP_STAGE_AUTH_TOKEN
+    * GCP_STAGE_AUTH_TOKEN_TRY
+    * GCP_PROD_AUTH_TOKEN
 
     \b
     Note: Due to Bug 1759740, we need separate auth tokens
     for try uploads with "Upload Try Symbols Files" permissions.
     """
-    destructive_tests = False
-    bad_token_test = False
-    tecken_host = ""
-    auth_token = ""
-
-    if env == "local":
-        destructive_tests = True
-        bad_token_test = True
-        tecken_host = "http://web:8000/"
-        auth_token = os.environ["LOCAL_AUTH_TOKEN"]
-    elif env == "stage":
-        destructive_tests = True
-        bad_token_test = True
-        tecken_host = "https://symbols.stage.mozaws.net/"
-        auth_token = os.environ["STAGE_AUTH_TOKEN"]
-    elif env == "prod":
-        click.echo(
-            click.style(
-                "Running tests in prod--skipping destructive tests!", fg="yellow"
-            )
-        )
-        tecken_host = "https://symbols.mozilla.org/"
-        auth_token = os.environ["PROD_AUTH_TOKEN"]
-    else:
-        raise click.UsageError("ENV must be one of 'local', 'stage', or 'prod'.")
+    try:
+        env = ENVIRONMENTS[env_name]
+    except KeyError:
+        env_names = ", ".join(f"'{name}'" for name in ENVIRONMENTS)
+        raise click.UsageError(f"ENV_NAME must be one of {env_names}.") from None
 
-    click.echo(click.style(f"tecken_host: {tecken_host}\n", fg="yellow"))
+    click.echo(click.style(f"Tecken base URL: {env.base_url}\n", fg="yellow"))
 
-    if destructive_tests:
+    if env.destructive_tests:
         click.echo(click.style(">>> UPLOAD TEST (DESTRUCTIVE)", fg="yellow"))
-        for fn in [
-            name
-            for name in os.listdir(f"{ZIPS_DIR}")
-            if os.path.isfile(f"{ZIPS_DIR}{name}")
-        ]:
-            # Use the try upload API token when needed
-            if fn.endswith("__try.zip"):
-                auth_token = os.environ[f"{env.upper()}_AUTH_TOKEN_TRY"]
-            else:
-                auth_token = os.environ[f"{env.upper()}_AUTH_TOKEN"]
+        for path in ZIP_FILES:
+            try_storage = path.stem.endswith("__try")
+            auth_token = env.auth_token(env_name, try_storage)
             ctx.invoke(
                 upload_symbols,
                 expect_code=201,
                 auth_token=auth_token,
-                base_url=tecken_host,
-                symbolsfile=f"{ZIPS_DIR}{fn}",
+                base_url=env.base_url,
+                symbolsfile=str(path),
             )
         click.echo("")
 
         click.echo(
             click.style(">>> UPLOAD BY DOWNLOAD TEST (DESTRUCTIVE)", fg="yellow")
         )
-        url = ctx.invoke(list_firefox_symbols_zips, number=1, max_size=1000000000)[0]
+        url = ctx.invoke(list_firefox_symbols_zips, number=1, max_size=1_000_000_000)[0]
         ctx.invoke(
             upload_symbols_by_download,
-            base_url=tecken_host,
+            base_url=env.base_url,
             auth_token=auth_token,
```

Comment:
`auth_token` has the potential to be undefined here because it's set inside a for loop above

Reply:
The presumption is that you've run `./setup_tests.py` prior to running this script, so that there are some `ZIP_FILES`, but we don't check that it's empty here. We could add a check to do so and remind people to setup the tests first?